### ISSUE TITLE
Semantic: anonymous container in a field type corrupts the parent container symbol flags

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#367]
# Semantic: anonymous container in a field type corrupts the parent container's symbol flags

- **Severity:** High (default-on `unused-decls` false positive; dangerous autofix deletes used code)
- **Status:** Verified by reproduction
- **Location:** `src/Semantic/Builder.zig:578` (`visitContainer`) + `src/Semantic/Builder.zig:662-665` (`visitContainerField`)

## The bug

`visitContainer` unconditionally sets container-kind flags on whatever symbol is on top of the symbol stack:

```zig
// Builder.zig:578
self.currentContainerSymbolFlags().set(symbol_flags, true);
```

For a *named* nested container (`const Kind = enum {...}`), the var-decl symbol was pushed by `visitVarDecl`, so the flags land on the right symbol. But an **anonymous** container appearing as a field's type pushes no symbol — the top of the stack is still the *enclosing* container. So `e: enum { on, off }` inside `struct Foo` sets `s_enum` on **Foo's** symbol.

`visitContainerField` then consults the parent's flags to decide whether to visit field type expressions:

```zig
// Builder.zig:662
const flags: []const Symbol.Flags = self.symbolTable().symbols.items(.flags);
if (!flags[parent].s_enum) {
    try self.visitOptionalType(field.ast.type_expr);
}
```

Once Foo is falsely marked `s_enum`, every field of Foo declared *after* the anonymous-enum field has its type expression skipped entirely — no references are recorded for types used there. Secondary damage: the anonymous enum's variants are `declareMemberSymbol`'d onto Foo's member list, and Foo carrying both `s_struct` and `s_enum` can misclassify later fields.

## User impact

Types referenced only in later field types are reported as unused by the default-on `unused-decls` rule, and the rule's autofix (applied by plain `--fix` per report 04) deletes them — breaking the build. Anonymous enum/struct/union field types are common idiomatic Zig.

## Reproduction

```sh
cat > sem2.zig <<'EOF'
const MyType = u32;
const Foo = struct {
    e: enum { on, off },   // poisons Foo's symbol with s_enum
    val: MyType,           // type never visited -> no reference to MyType
};
pub fn main() void {
    _ = Foo{ .e = .on, .val = 1 };
}
EOF
zlint sem2.zig
# warning: unused-decls: variable 'MyType' is declared but never used.
```

## Proposed fix

`visitContainer` must only stamp container-kind flags when the container is the initializer of the symbol on top of the stack (i.e. when the node being visited is the var decl's init node), not for arbitrary nested/anonymous containers. Options:

1. Track the "current symbol's initializer node" and compare: only call `currentContainerSymbolFlags().set(...)` when `node_id` matches it.
2. For anonymous containers, push a fresh anonymous symbol (or a null sentinel) so both the flag write and the member declarations (`declareMemberSymbol`) target the anonymous container rather than the enclosing one. This also fixes the variants-attached-to-wrong-container corruption.

Option 2 is more complete. Add regression tests for anonymous container field types followed by typed fields (the existing tests around named nested containers, `symbol_decl_test.zig:345-380`, don't cover this).
